src/app/account/invitation/[inviteId]/invitation-handler.tsx

```diff
@@ -0,0 +1,128 @@
+    // use dynamic dashboard resolution
+    const dashboardUrl = useDashboardUrl()
+
+    // remember the inviteId so that once they sign in we auto-claim
+    useEffect(() => {
+        window.localStorage.setItem('pendingInviteId', inviteId)
+    }, [inviteId])
+
+    useEffect(() => {
+        if (isLoaded && isSignedIn) {
+            setIsLoadingAction(true)
+            claimInviteAction({ inviteId })
+                .then((result) => {
+                    if (result.success && result.organizationName) {
+                        setOrgName(result.organizationName)
+                        setNewOrgSlug(result.orgSlug || null)
+                        // The success message is now part of the SuccessPanel
+                    } else {
+                        setError(result.error || 'Failed to accept invitation.')
+                        notifications.show({
+                            title: 'Error',
+                            message: result.error || 'Could not process your invitation.',
+                            color: 'red',
+                        })
+                    }
+                })
+                .catch((err) => {
+                    reportError(err, 'Error accepting invitation for existing user')
+                    setError('An unexpected error occurred while processing your invitation.')
+                    notifications.show({
+                        title: 'Error',
+                        message: 'An unexpected error occurred.',
+                        color: 'red',
+                    })
+                })
+                .finally(() => setIsLoadingAction(false))
+        }
+    }, [isLoaded, isSignedIn, inviteId])
+
+    useEffect(() => {
+        if (!newOrgSlug || !user) return
+
+        const orgMeta = user.publicMetadata?.orgs?.find((o) => o.slug === newOrgSlug)
+        if (orgMeta?.isReviewer) {
+            // update Clerk’s active organization so useDashboardUrl picks it up
+            setActive({ organization: newOrgSlug }).catch(() => {})
+        }
+    }, [newOrgSlug, user, setActive])
+
+    // fetch whether this email is already in our SI user table
+    const { data: userExists } = useQuery({
+        queryKey: ['userExists', invitedEmail],
+        queryFn: () => userExistsAction(invitedEmail),
+        enabled: isLoaded,
+    })
+
+    if (!isLoaded || isLoadingAction) {
+        return <LoadingMessage message="Processing your invitation..." />
+    }
+
+    // returning user: prompt them to sign in instead of sign up
+    if (userExists && !isSignedIn) {
+        return <SignIn />
+    }
+
+    if (isSignedIn) {
+        if (error) {
+            return <Text color="red">{error}</Text>
+        }
+        if (orgName) {
+            return (
+                <SuccessPanel title={`You're now a member of ${orgName}!`} onContinue={() => router.push(dashboardUrl)}>
+                    Visit your dashboard
+                </SuccessPanel>
+            )
```

Comment:
do you know the user exists at this point? I may be misreading all the if statements, but I don't think you're checking that here